dimod/sampleset.py

```diff
@@ -580,6 +574,16 @@ def __eq__(self, other):
 
         return (self.record.sample == other.record.sample[:, other_idx]).all()
 
+    def __getstate__(self):
+        """Ensure that any futures are resolved before pickling."""
+        self.resolve()
+        return {attr: getattr(self, attr)
+                for attr in self.__slots__ if hasattr(self, attr)}
+
+    def __setstate__(self, state):
+        for attr, obj in state.items():
+            setattr(self, attr, obj)
```

Comment:
Could you just defer pickling to super methods, after the future is resolved?

Reply:
I tried that, but because we're using `__slots__` there is no `super().__getstate__`. We could maybe use `__reduce__` but that's getting a bit deep and I wasn't able to get it to work in a way I felt confident about edge cases

Reply:
Also to support python2 format which doesn't support `__slots__` at all.

Reply:
I see. OK, let's go with this than.